libs/core/src/lib/action-sheet/action-sheet.component.ts

```diff
@@ -0,0 +1,101 @@
+import {
+    AfterContentInit,
+    ChangeDetectionStrategy,
+    Component, ContentChildren, ElementRef,
+    EventEmitter, Injector,
+    Input,
+    Optional, Output, QueryList, TemplateRef, ViewChild,
+    ViewEncapsulation
+} from '@angular/core';
+import { BehaviorSubject } from 'rxjs';
+import { Placement } from 'popper.js';
+import { DynamicComponentService } from '../utils/dynamic-component/dynamic-component.service';
+import { ActionSheetMobileComponent } from './action-sheet-mobile/action-sheet-mobile.component';
+import { ACTION_SHEET_COMPONENT, ActionSheetInterface } from './action-sheet.interface';
+import {ActionSheetItemComponent} from './action-sheet-item/action-sheet-item.component';
+
+
+@Component({
+    selector: 'fd-action-sheet',
+    templateUrl: './action-sheet.component.html',
+    styleUrls: ['./action-sheet.component.scss'],
+    encapsulation: ViewEncapsulation.None,
+    changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class ActionSheetComponent implements AfterContentInit, ActionSheetInterface {
+
+    constructor(
+        private _elementRef: ElementRef,
+        @Optional() private _dynamicComponentService: DynamicComponentService
+    ) {}
+
+    @ContentChildren(ActionSheetItemComponent) actionSheetItems: QueryList<ActionSheetItemComponent>;
```

Comment:
This file should follow order from [Code Contribution Guideline](https://github.com/SAP/fundamental-ngx/wiki/Code-Contribution-Guideline)